aptos-move/framework/aptos-framework/sources/configs/Stake.move

```diff
@@ -98,124 +108,275 @@ module AptosFramework::Stake {
-        if (!is_current_validator(to)) {
-            IterableTable::add(&mut stake_pool.active, addr, delegation);
-        } else {
-            IterableTable::add(&mut stake_pool.pending_active, addr, delegation);
-        }
+    /// TestCoin capabilities, set during genesis and stored in @CoreResource account.
+    /// This allows the Stake module to mint rewards to stakers.
+    struct TestCoinCapabilities has key {
+        mint_cap: MintCapability<TestCoin>,
     }
 
-    /// Unlock from active delegation, it's moved to pending_inactive if locked_until_secs < current_time or
-    /// directly inactive if it's not from an active validator.
-    public(script) fun unlock(account: &signer, from: address) acquires StakePool, ValidatorSet {
-        let addr = Signer::address_of(account);
-        let current_time = Timestamp::now_seconds();
-        let stake_pool = borrow_global_mut<StakePool>(from);
-        let delegation = withdraw_internal(&mut stake_pool.active, addr);
-        let is_current_validator = is_current_validator(from);
-        stake_pool.next_epoch_voting_power = stake_pool.next_epoch_voting_power - Coin::value<TestCoin>(&delegation.coins);
-        if (!is_current_validator) {
-            // move to inactive directly if it's not from an active validator
-            IterableTable::add(&mut stake_pool.inactive, addr, delegation);
-        } else if (delegation.locked_until_secs < current_time) {
-            // move to pending_inactive if it can be unlocked
-            IterableTable::add(&mut stake_pool.pending_inactive, addr,  delegation);
-        } else {
-            abort Errors::invalid_argument(EWITHDRAW_NOT_ALLOWED)
-        };
+    /// Initialize validator set to the core resource account.
+    public(friend) fun initialize_validator_set(account: &signer, minimum_stake: u64, maximum_stake: u64) {
+        SystemAddresses::assert_core_resource(account);
+        move_to(account, ValidatorSet {
+            consensus_scheme: 0,
+            minimum_stake,
+            maximum_stake,
+            active_validators: Vector::empty(),
+            pending_active: Vector::empty(),
+            pending_inactive: Vector::empty(),
+        });
     }
 
-    /// Withdraw from inactive delegation.
-    public(script) fun withdraw(account: &signer, from: address): Coin<TestCoin> acquires StakePool {
-        let addr = Signer::address_of(account);
-        let stake_pool = borrow_global_mut<StakePool>(from);
-        let delegation = withdraw_internal(&mut stake_pool.inactive, addr);
-        let Delegation {coins, rewards, locked_until_secs: _} = delegation;
-        Coin::merge<TestCoin>(&mut coins, rewards);
-        coins
+    /// This is only called during Genesis, which is where MintCapability<TestCoin> can be created.
+    /// Beyond genesis, no one can create TestCoin mint/burn capabilities.
+    public(friend) fun store_test_coin_mint_cap(core_resources: &signer, mint_cap: MintCapability<TestCoin>) {
```

Comment:
considering the requirements for this function to be executed, (friend) seems unnecessary.

Reply:
Updated